### PR TITLE
Fix OS release information in system details

### PR DESCRIPTION
**Pull request by achawla2012**
### Proposed changes

Closes https://github.com/nginx/agent/issues/224

Update releaseInfo() function in environment.go. 

As part of this update, 
- Pull release-info from `os_release ` file  from the host if present. 
- Read information such as "VERSION_ID" "VERSION" "VERSION_CODENAME" "NAME" "ID"  from release file, if not present fall back to read it from the host info.

` {'mountpoint': '/snap/core18/2701', 'device': '/dev/loop9', 'fstype': 'squashfs'}], 'release': {'name': 'Ubuntu', 'version_id': '20.04', 'version': '20.04.5 LTS (Focal Fossa)', 'codename': 'focal', 'id': 'ubuntu'},`

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [X] If applicable, I have added tests that prove my fix is effective or that my feature works
- [X] If applicable, I have checked that any relevant tests pass after adding my changes
- [X] I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))

#### Test  -
- Added Units tests for couple of os-release file with different content.
- Deployed agent changes in sandbox. Validated change by intercepting message that converter sends to receiver (aka AMP backend).

### Ubuntu 22

```
ubuntu@ip-172-16-0-49:~$ cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

```
//agent .log
"{\"agent\":\"\",\"boot\":1677788139,\"hostname\":\"ip-172-16-0-49\",\"display_name\":\"ip-172-16-0-49\",\"os-type\":\"linux\",\"uuid\":\"1325a622-f367-3f34-b907-a85d183844e8\",\"uname\":\"Linux ip-172-16-0-49 5.15.0-1030-aws #34-Ubuntu SMP Mon Jan 23 20:13:32 UTC 2023 x86_64\",\"disk_partitions\":[{\"mountpoint\":\"/\",\"device\":\"/dev/nvme0n1p1\",\"fstype\":\"ext4\"},{\"mountpoint\":\"/snap/amazon-ssm-agent/6312\",\"device\":\"/dev/loop0\",\"fstype\":\"squashfs\"},{\"mountpoint\":\"/snap/core20/1822\",\"device\":\"/dev/loop2\",\"fstype\":\"squashfs\"},{\"mountpoint\":\"/snap/snapd/18357\",\"device\":\"/dev/loop4\",\"fstype\":\"squashfs\"},{\"mountpoint\":\"/snap/lxd/24322\",\"device\":\"/dev/loop3\",\"fstype\":\"squashfs\"},{\"mountpoint\":\"/snap/core18/2679\",\"device\":\"/dev/loop1\",\"fstype\":\"squashfs\"},{\"mountpoint\":\"/boot/efi\",\"device\":\"/dev/nvme0n1p15\",\"fstype\":\"vfat\"},{\"mountpoint\":\"/snap/core18/2697\",\"device\":\"/dev/loop5\",\"fstype\":\"squashfs\"}],\"network\":{\"interfaces\":[{\"mac\":\"\",\"ipv6\":[{\"prefixlen\":128,\"netmask\":\"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff\",\"address\":\"::1\"}],\"ipv4\":[{\"prefixlen\":8,\"netmask\":\"255.0.0.0\",\"address\":\"127.0.0.1\"}],\"name\":\"lo\"},{\"mac\":\"0a:2b:71:5b:33:33\",\"ipv6\":[{\"prefixlen\":128,\"netmask\":\"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff\",\"address\":\"2600:1f13:a98:5701:a8ea:8a0e:327:a8b7\"},{\"prefixlen\":64,\"netmask\":\"ffff:ffff:ffff:ffff::\",\"address\":\"fe80::82b:71ff:fe5b:3333\"}],\"ipv4\":[{\"prefixlen\":24,\"netmask\":\"255.255.255.0\",\"address\":\"172.16.0.49\"}],\"name\":\"ens5\"},{\"mac\":\"02:42:af:f1:90:b4\",\"ipv6\":[],\"ipv4\":[{\"prefixlen\":24,\"netmask\":\"255.255.255.0\",\"address\":\"192.0.10.1\"}],\"name\":\"docker0\"}],\"default\":\"lo\"},\"processor\":[{\"model\":\"1\",\"cores\":1,\"architecture\":\"x86_64\",\"mhz\":2200,\"hypervisor\":\"\",\"cpus\":2,\"virtualization\":\"\",\"cache\":{\"Cacheline bytes:\":\"64\",\"Features:\":\"ADX,AESNI,AVX,AVX2,AVXSLOW,BMI1,BMI2,CLMUL,CLZERO,CMOV,CMPXCHG8,CX16,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MMXEXT,MOVBE,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SHA,SSE,SSE2,SSE3,SSE4,SSE42,SSE4A,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT\",\"L1d\":\"32 KiB (1 instance)\",\"L1i\":\"64 KiB (1 instance)\",\"L2\":\"512 KiB (1 instance)\",\"L3\":\"8 MiB (1 instance)\",\"SIMD 2:\":\"Streaming SIMD 2 Extensions\"}},{\"model\":\"1\",\"cores\":1,\"architecture\":\"x86_64\",\"mhz\":2200,\"hypervisor\":\"\",\"cpus\":2,\"virtualization\":\"\",\"cache\":{\"Cacheline bytes:\":\"64\",\"Features:\":\"ADX,AESNI,AVX,AVX2,AVXSLOW,BMI1,BMI2,CLMUL,CLZERO,CMOV,CMPXCHG8,CX16,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MMXEXT,MOVBE,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SHA,SSE,SSE2,SSE3,SSE4,SSE42,SSE4A,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT\",\"L1d\":\"32 KiB (1 instance)\",\"L1i\":\"64 KiB (1 instance)\",\"L2\":\"512 KiB (1 instance)\",\"L3\":\"8 MiB (1 instance)\",\"SIMD 2:\":\"Streaming SIMD 2 Extensions\"}}],\"release\":{\"codename\":\"jammy\",\"id\":\"ubuntu\",\"name\":\"Ubuntu\",\"version_id\":\"22.04\",\"version\":\"22.04.2 LTS (Jammy Jellyfish)\"},\"tags\":[],\"agent_accessible_dirs\":\"/etc/nginx:/usr/local/etc/nginx:/usr/share/nginx/modules:/etc/nms\"}"
```

### RHEL

```
[ec2-user@ip-172-16-0-214 ~]$ cat /etc/os-release 
NAME="Red Hat Enterprise Linux"
VERSION="8.7 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.7"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.7 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/red_hat_enterprise_linux/8/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.7
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.7" 
```

```
// Agent log.
"{\"agent\":\"\",\"boot\":1677651430,\"hostname\":\"ip-172-16-0-214.us-west-2.compute.internal\",\"display_name\":\"ip-172-16-0-214.us-west-2.compute.internal\",\"os-type\":\"linux\",\"uuid\":\"42721515-dc58-381e-b208-b806af379e51\",\"uname\":\"Linux ip-172-16-0-214.us-west-2.compute.internal 4.18.0-425.10.1.el8_7.x86_64 #1 SMP Wed Dec 14 16:00:01 EST 2022 x86_64\",\"disk_partitions\":[{\"mountpoint\":\"/\",\"device\":\"/dev/nvme0n1p2\",\"fstype\":\"xfs\"}],\"network\":{\"interfaces\":[{\"mac\":\"\",\"ipv6\":[{\"prefixlen\":128,\"netmask\":\"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff\",\"address\":\"::1\"}],\"ipv4\":[{\"prefixlen\":8,\"netmask\":\"255.0.0.0\",\"address\":\"127.0.0.1\"}],\"name\":\"lo\"},{\"mac\":\"02:3d:1b:71:ac:af\",\"ipv6\":[{\"prefixlen\":128,\"netmask\":\"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff\",\"address\":\"2600:1f13:553:8001:9426:65e7:ad32:c6b0\"},{\"prefixlen\":64,\"netmask\":\"ffff:ffff:ffff:ffff::\",\"address\":\"fe80::3d:1bff:fe71:acaf\"}],\"ipv4\":[{\"prefixlen\":24,\"netmask\":\"255.255.255.0\",\"address\":\"172.16.0.214\"}],\"name\":\"eth0\"}],\"default\":\"lo\"},\"processor\":[{\"model\":\"1\",\"cores\":1,\"architecture\":\"x86_64\",\"mhz\":2199.522,\"hypervisor\":\"\",\"cpus\":2,\"virtualization\":\"\",\"cache\":{\"Cacheline bytes:\":\"64\",\"Features:\":\"ADX,AESNI,AVX,AVX2,AVXSLOW,BMI1,BMI2,CLMUL,CLZERO,CMOV,CMPXCHG8,CX16,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MMXEXT,MOVBE,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SHA,SSE,SSE2,SSE3,SSE4,SSE42,SSE4A,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT\",\"L1d\":\"32K\",\"L1i\":\"64K\",\"L2\":\"512K\",\"L3\":\"8192K\",\"SIMD 2:\":\"Streaming SIMD 2 Extensions\"}},{\"model\":\"1\",\"cores\":1,\"architecture\":\"x86_64\",\"mhz\":2199.522,\"hypervisor\":\"\",\"cpus\":2,\"virtualization\":\"\",\"cache\":{\"Cacheline bytes:\":\"64\",\"Features:\":\"ADX,AESNI,AVX,AVX2,AVXSLOW,BMI1,BMI2,CLMUL,CLZERO,CMOV,CMPXCHG8,CX16,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MMXEXT,MOVBE,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SHA,SSE,SSE2,SSE3,SSE4,SSE42,SSE4A,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT\",\"L1d\":\"32K\",\"L1i\":\"64K\",\"L2\":\"512K\",\"L3\":\"8192K\",\"SIMD 2:\":\"Streaming SIMD 2 Extensions\"}}],\"release\":{\"codename\":\"linux\",\"id\":\"rhel\",\"name\":\"Red Hat Enterprise Linux\",\"version_id\":\"8.7\",\"version\":\"8.7 (Ootpa)\"},\"tags\":[],\"agent_accessible_dirs\":\"/etc/nginx:/usr/local/etc/nginx:/usr/share/nginx/modules:/etc/nms\"}" 
```

### SUSE

```
ec2-user@ip-172-16-0-207:~> cat /etc/os-release 
NAME="SLES"
VERSION="15-SP2"
VERSION_ID="15.2"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP2"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp2"
```

```
//agent log
"{\"agent\":\"\",\"boot\":1677783989,\"hostname\":\"ip-172-16-0-207\",\"display_name\":\"ip-172-16-0-207\",\"os-type\":\"linux\",\"uuid\":\"d1af3bda-e1dd-351d-85e7-0784aa7f7fe0\",\"uname\":\"Linux ip-172-16-0-207 5.3.18-24.96-default #1 SMP Mon Nov 22 08:33:36 UTC 2021 (9a413cc) x86_64\",\"disk_partitions\":[{\"mountpoint\":\"/\",\"device\":\"/dev/nvme0n1p3\",\"fstype\":\"xfs\"},{\"mountpoint\":\"/boot/efi\",\"device\":\"/dev/nvme0n1p2\",\"fstype\":\"vfat\"}],\"network\":{\"interfaces\":[{\"mac\":\"\",\"ipv6\":[{\"prefixlen\":128,\"netmask\":\"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff\",\"address\":\"::1\"}],\"ipv4\":[{\"prefixlen\":8,\"netmask\":\"255.0.0.0\",\"address\":\"127.0.0.1\"}],\"name\":\"lo\"},{\"mac\":\"02:21:3a:31:34:d5\",\"ipv6\":[{\"prefixlen\":64,\"netmask\":\"ffff:ffff:ffff:ffff::\",\"address\":\"2600:1f13:843:6101:9691:3ac9:6f9:5a2d\"},{\"prefixlen\":64,\"netmask\":\"ffff:ffff:ffff:ffff::\",\"address\":\"fe80::21:3aff:fe31:34d5\"}],\"ipv4\":[{\"prefixlen\":24,\"netmask\":\"255.255.255.0\",\"address\":\"172.16.0.207\"}],\"name\":\"eth0\"},{\"mac\":\"02:42:56:7e:ca:ca\",\"ipv6\":[],\"ipv4\":[{\"prefixlen\":24,\"netmask\":\"255.255.255.0\",\"address\":\"192.0.10.1\"}],\"name\":\"docker0\"}],\"default\":\"lo\"},\"processor\":[{\"model\":\"1\",\"cores\":1,\"architecture\":\"x86_64\",\"mhz\":2199.79,\"hypervisor\":\"\",\"cpus\":2,\"virtualization\":\"\",\"cache\":{\"Cacheline bytes:\":\"64\",\"Features:\":\"ADX,AESNI,AVX,AVX2,AVXSLOW,BMI1,BMI2,CLMUL,CLZERO,CMOV,CMPXCHG8,CX16,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MMXEXT,MOVBE,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SHA,SSE,SSE2,SSE3,SSE4,SSE42,SSE4A,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT\",\"L1d\":\"32K\",\"L1i\":\"64K\",\"L2\":\"512K\",\"L3\":\"8192K\",\"SIMD 2:\":\"Streaming SIMD 2 Extensions\"}},{\"model\":\"1\",\"cores\":1,\"architecture\":\"x86_64\",\"mhz\":2199.79,\"hypervisor\":\"\",\"cpus\":2,\"virtualization\":\"\",\"cache\":{\"Cacheline bytes:\":\"64\",\"Features:\":\"ADX,AESNI,AVX,AVX2,AVXSLOW,BMI1,BMI2,CLMUL,CLZERO,CMOV,CMPXCHG8,CX16,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MMXEXT,MOVBE,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SHA,SSE,SSE2,SSE3,SSE4,SSE42,SSE4A,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT\",\"L1d\":\"32K\",\"L1i\":\"64K\",\"L2\":\"512K\",\"L3\":\"8192K\",\"SIMD 2:\":\"Streaming SIMD 2 Extensions\"}}],\"release\":{\"codename\":\"linux\",\"id\":\"sles\",\"name\":\"SLES\",\"version_id\":\"15.2\",\"version\":\"15-SP2\"},\"tags\":[],\"agent_accessible_dirs\":\"/etc/nginx:/usr/local/etc/nginx:/usr/share/nginx/modules:/etc/nms\"}"
```

### FreeBSD

```
ec2-user@ip-172-16-0-190:~ $ sudo cat /etc/os-release 
NAME=FreeBSD
VERSION="13.1-RELEASE-p7"
VERSION_ID="13.1"
ID=freebsd
ANSI_COLOR="0;31"
PRETTY_NAME="FreeBSD 13.1-RELEASE-p7"
CPE_NAME="cpe:/o:freebsd:freebsd:13.1"
HOME_URL="https://FreeBSD.org/"
BUG_REPORT_URL="https://bugs.FreeBSD.org/"
ec2-user@ip-172-16-0-190:~ $ 
```

As I spined freebsd instance on testenv with datapath_os = `freebsd`, I am seeing lot of error in `agent.log` ( without my changes..) 
I would still see the errors even if I create another new instance. 

It's been flagged here https://github.com/nginx/agent/blob/960eefb7c7649dc2b36b15a2480b644c1f23e4be/src/core/nginx.go#L222

```
time="2023-03-06T23:02:42Z" level=warning msg="Unable to find the correct NGINX binary in $PATH: exec: \"nginx\": executable file not found in $PATH"
time="2023-03-06T23:02:42Z" level=error msg="Error reading exe information for process: 2051 error: not implemented yet"
```

```
ec2-user@ip-172-16-0-190:~ $  /usr/local/bin/nginx-agent -v
nginx-agent version v2.23.1-60ae137d
ec2-user@ip-172-16-0-190:~ $ ps aux | grep nginx
root      2051   0.0  0.2  19588  8816  -  Is   08:48      0:00.00 nginx: master process /usr/local/sbin/nginx
www       2052   0.0  0.2  19588  9276  -  I    08:48      0:00.00 nginx: worker process (nginx)
www       2053   0.0  0.2  19588  9272  -  I    08:48      0:00.00 nginx: worker process (nginx)
root      2384   0.0  0.1  12848  2140  -  Ss   08:51      0:03.20 daemon: /usr/local/bin/nginx-agent[2385] (daemon)
root      2385   0.0  1.2 729604 47588  -  S    08:51     35:44.33 /usr/local/bin/nginx-agent
ec2-user 30095   0.0  0.1  12920  2312  0  S+   23:03      0:00.00 grep nginx
ec2-user@ip-172-16-0-190:~ $ 
```

Agent log
```
time="2023-03-14T14:54:56Z" level=trace msg="HostInfo created: boot:1678725802 hostname:\"ip-172-16-0-163.eu-central-1.compute.internal\" display_name:\"ip-172-16-0-163.eu-central-1.compute.internal\" os_type:\"freebsd\" uuid:\"80726cf1-f310-3136-9438-f707908a74ef\" uname:\"amd64\" partitons:<mount_point:\"/\" device:\"/dev/gpt/rootfs\" fs_type:\"ufs\" > partitons:<mount_point:\"/dev\" device:\"devfs\" fs_type:\"devfs\" > partitons:<mount_point:\"/boot/efi\" device:\"/dev/gpt/efiesp\" fs_type:\"msdosfs\" > network:<interfaces:<mac:\"06:a2:78:8e:a5:a8\" ipv6:<prefixlen:64 netmask:\"ffff:ffff:ffff:ffff::\" address:\"fe80::4a2:78ff:fe8e:a5a8\" > ipv4:<prefixlen:24 netmask:\"255.255.255.0\" address:\"172.16.0.163\" > name:\"ena0\" > interfaces:<ipv6:<prefixlen:128 netmask:\"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff\" address:\"::1\" > ipv6:<prefixlen:64 netmask:\"ffff:ffff:ffff:ffff::\" address:\"fe80::1\" > ipv4:<prefixlen:8 netmask:\"255.0.0.0\" address:\"127.0.0.1\" > name:\"lo0\" > default:\"ena0\" > processor:<cores:2 mhz:2500 cpus:1 virtualization:\"host\" cache:<key:\"Cacheline bytes:\" value:\"64\" > cache:<key:\"Features:\" value:\"ADX,AESNI,AVX,AVX2,AVX512BW,AVX512CD,AVX512DQ,AVX512F,AVX512VL,BMI1,BMI2,CLMUL,CMOV,CMPXCHG8,CX16,ERMS,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MOVBE,MPX,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SSE,SSE2,SSE3,SSE4,SSE42,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT,XSAVES\" > cache:<key:\"L1d\" value:\"32768\" > cache:<key:\"L1i\" value:\"32768\" > cache:<key:\"L2\" value:\"1048576\" > cache:<key:\"L3\" value:\"37486592\" > cache:<key:\"SIMD 2:\" value:\"Streaming SIMD 2 Extensions\" > > release:<codename:\"freebsd\" id:\"freebsd\" name:\"FreeBSD\" version_id:\"13.1\" version:\"13.1-RELEASE-p7\" > agent_accessible_dirs:\"/etc/nginx:/usr/local/etc/nginx:/usr/share/nginx/modules:/etc/nms\" "
```

### CODE REVIEW

**Comment by achawla2012** [comment]
@oliveromahony

File: src/core/environment_test.go [hunk 1]
```diff
@@ -10,13 +10,51 @@ package core
...
+const ubuntuReleaseInfo = `
+NAME="Ubuntu"
+VERSION="20.04.5 LTS (Focal Fossa)"
+VERSION_ID="20.04"
+ID=ubuntu
+ID_LIKE=debian
+PRETTY_NAME="Ubuntu 20.04.5 LTS"
+HOME_URL="https://www.ubuntu.com"
+SUPPORT_URL=\"https://help.ubuntu.com/"
+BUG_REPORT_URL=\"https://bugs.launchpad.net/ubuntu/"
```
Comment (ryepup) [inline src/core/environment_test.go:52]: do we need these escapes?
```suggestion
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
```

**Review by ryepup — COMMENTED** [review]
LGTM

File: src/core/environment_test.go (hunk 1 above)
Comment (achawla2012) [reply src/core/environment_test.go:52]: I do not recall adding these  escapes explicitly. I have nevertheless got rid of it. Thanks

**Review by aphralG — APPROVED** [review]

**Review by dhurley — APPROVED** [review]

**Review by oliveromahony — APPROVED** [review]

**Pull request merged**